### ISSUE TITLE
image size

### ISSUE THREAD

**Issue by Ehteshamciitwah** [dahyun-kang/ifsl#10]
Hello, Thank you for your work. Is the model suitable for all image sizes? I check with image sizes (256x256) and (384x384). But it shows errors or padding and dimensions. Can you please elaborate the relation of image size with model structure? Thank you

**Comment by dahyun-kang** [comment]
Hi

In the current implementation, the kernel sizes, padding, and stride are fit to the 400x400 image resolution which returns 13x13, 25x25, 50x50 feature map sizes from ResNet intermediate layers.
But you could also make adjustments by yourself on the sizes at 
https://github.com/dahyun-kang/ifsl/blob/eb2cf9604ff5d55e3a158d25ebc27fcb32ada067/fs-cs/model/module/asnet.py#L14
https://github.com/dahyun-kang/ifsl/blob/eb2cf9604ff5d55e3a158d25ebc27fcb32ada067/fs-cs/model/module/asnet.py#L23-L25
As you want to reduce the image size, I'd recommend you reduce the kernel size and strides and find the right sizes.
Have a good day! :smiley: 

Best,
Dahyun

**Comment by Ehteshamciitwah** [comment]
Thank you for your quick response. I will try to change it for image size (W,H)/(768,256). Do you have a pre-trained model for custom inputs or we need to train from scratch if I want to use other image sizes than 400? Thanks.

**Comment by dahyun-kang** [comment]
Unfortunately, we do not have model checkpoints trained on other image sizes readily available.

**Comment by Ehteshamciitwah** [comment]
hmmm. Thanks for your responses. I have a last question. Is it possible to use one model for different image sizes or does one need to retrain every time for different image size? Thanks

**Comment by dahyun-kang** [comment]
The model gradually resizes the predefined input sizes in a fixed strided fashion, thus a model trained image size A cannot be transfered to perform with image size B as is.
There are two options I would suggest: 1) adjusting the model kernel&strides and retraining it from scratch with a certain image size or 2) resize the input correlation to fit the pretrained checkpoints.

**Comment by Ehteshamciitwah** [comment]
Hello,as per your suggestion, I resize the correlation tensor to default before the encoder and use pre-trained weights. But after 150 epochs the miou is just 40.

**Comment by dahyun-kang** [comment]
Yes, a lower resolution feature would obviously have less sophisticated details and hence lower segmentation performance :slightly_smiling_face:
